array/slice.mbt

```diff
@@ -0,0 +1,47 @@
+// Copyright 2024 International Digital Economy Academy
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+struct ArraySlice[T] {
```

Comment:
Should we mark this as `priv` to avoid exposing this struct to users?

Reply:
I initially thought you wanted users to use this structure, and I think slice is a nice feature for arrays. However, there is no standardized IterableTrait for iterable structures yet, so do we consider slice for all iterable structures at that time?

Reply:
Yes, slices are useful. But I'm unsure whether this is the best way to do it because it doesn't work on Vec right now. I think there will be a better interface for slicing when `Trait[T]` is available. 

Slice cannot work on all iterable structures. If I understand your question correctly, we can use `.skip(n)` and `.take(n)` to achieve a similar effect for the iterator.

Reply:
对的，实际上现在的trait还没做到多态，Self的写法限制太多了，所以写不出迭代的trait这个局面还蛮尴尬的。但是这个问题也一样存在在这些数据结构的函数上面，现在大家都是一个结构实现一套，所以现在slice其实加过来也无伤大雅。不过长远考虑的话我还是把它先private了，以上。

Right, in fact, trait is not polymorphic yet, and Self is too limited to write, so it is quite awkward to write an iterative trait. But this problem also exists in the function of these data structures, and now everyone is implementing a set of structures, so now slice actually adds up to no harm. But I'm gonna keep it private for the long haul. Above.

Reply:
Agree